spec/lib/granite/rubocop/cop/performer_in_initializer_spec.rb

```diff
@@ -0,0 +1,250 @@
+require 'rubocop/rspec/support'
+require 'granite/rubocop/cop/performer_in_initializer'
+
+RSpec.describe RuboCop::Cop::Granite::PerformerInInitializer do
+  subject(:cop) { described_class.new }
+
+  let(:message) { described_class::MSG }
+
+  before do
+    inspect_source(source)
+  end
+
+  shared_examples 'code without offense' do |code|
+    let(:source) { code }
+
+    it 'does not register an offense' do
+      expect(cop.offenses).to be_empty
+    end
+  end
+
+  shared_examples 'code with offense' do |code, corrected = nil|
+    let(:source) { code }
+
+    it 'registers an offense' do
+      expect(cop.offenses.size).to eq(1)
+      expect(cop.messages).to eq([message])
+    end
+
+    if corrected
+      it 'auto-corrects' do
+        expect(autocorrect_source(code)).to eq(corrected)
+      end
+    else
+      it 'does not auto-correct' do
+        expect(autocorrect_source(code)).to eq(code)
+      end
+    end
+  end
+
+  context 'when is not a real Business Action' do
+    it_behaves_like 'code without offense',
+                    'NotBusinessAction.new(performer: Role.system)'
+
+    it_behaves_like 'code without offense',
+                    'NotBA::Subject::Action.new(performer: Role.system)'
+
+    it_behaves_like 'code without offense',
+                    '5.inspect'
+  end
+
+  context 'when already using the new syntax' do
+    code = 'BA::TshirtOrder::Create.as(attributes.delete(:performer)).new(attributes)'
+    it_behaves_like 'code without offense', code
+  end
+
+  context 'when checking specs using `described_class`' do
+    context 'with role system' do
+      code = <<-CODE
+      describe BA::Subject::Action do
+        subject { described_class.new(performer: Role.system) }
+      end
+      CODE
+      corrected = <<-CORRECTED
+      describe BA::Subject::Action do
+        subject { described_class.as_system.new() }
+      end
+      CORRECTED
+      it_behaves_like 'code with offense', code, corrected
```

Comment:
I think the new wave of cop testing suggests using `expect_offense`/`expect_correction`.